integrations/gitlab/gitlab_integration/gitlab_service.py

```diff
@@ -753,58 +751,100 @@ async def enrich_object_with_members(
         obj: RESTObject,
         include_inherited_members: bool = False,
         include_bot_members: bool = True,
+        batch_size: int = 50,
+        include_verbose_member_object: bool = False,
     ) -> RESTObject:
         """
-        Enriches an object (e.g., Project or Group) with its members and optionally their public emails.
+        Enriches an object (e.g., Project or Group) with its members.
         """
-        members_list = [
-            member
-            async for members in self.get_all_object_members(
+        obj_name = getattr(obj, "name", "unknown")
+        logger.info(f"Starting member enrichment for {obj_name}")
+
+        setattr(obj, "__members", [])
+        members_list = getattr(obj, "__members")
+
+        processed_member_ids: Set[int] = set()
+        total_members_processed = 0
+
+        try:
+            async for members_batch in self.get_all_object_members(
                 obj, include_inherited_members, include_bot_members
+            ):
+                for member_chunk_idx in range(0, len(members_batch), batch_size):
+                    member_chunk = members_batch[
+                        member_chunk_idx : member_chunk_idx + batch_size
+                    ]
+
+                    # Filter out duplicates
+                    for member in member_chunk:
+                        member_id = getattr(member, "id", None)
+                        if (
+                            member_id is not None
+                            and member_id not in processed_member_ids
+                        ):
+                            processed_member_ids.add(member_id)
+                            if include_verbose_member_object:
+                                members_list.append(member.asdict())
+                            else:
+                                members_list.append(
+                                    {
+                                        "id": member.id,
+                                        "username": member.username,
+                                        "email": getattr(member, "email", ""),
+                                    }
+                                )
+                    total_members_processed += len(member_chunk)
+
+                logger.info(
+                    f"Processed {total_members_processed} members for {obj_name}"
+                )
```

Comment:
why you are chunking the members? seems like you can iteratre over the whole member batch and just add it to the member list